src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -252,6 +252,7 @@ void mf::WlSurface::commit(WlSurfaceState const& state)
 
 void mf::WlSurface::commit()
 {
+    commit(pending);
```

Comment:
There are good reasons I structured it so that WlSurfaceRole::commit() calls this. The main one being that subsurfaces must intercept the state and apply it later.

Reply:
Regardless of your "good reasons", commit() can not be a no-op when no role has been set.

It should, for example, submit the buffer even when no role has been set. (See this PR for one reason.)

Let's meet on IRC?